torch/csrc/jit/fuser/executor.cpp

```diff
@@ -339,6 +344,12 @@ bool runFusion(const int64_t key, Stack& stack) {
   std::vector<at::Tensor> outputs;
   launchFusion(*(*maybe_kernel), device, inputs, outputs);
 
+  for (size_t i = 0; i < outputs.size(); i++) {
```

Comment:
I don't know if it affects perf a lot, but at this point with runtime shapes information you already know when in fact there are no reductions required, so this at::sum_to can be skipped altogether.

Reply:
I would hope that sum_to is fast if there isn't anything to be done, even if it compares number of tensors * average dimension sizes.

What would be the condition to skip the entire thing?

Reply:
It should be fast, it's just I've been seeing cases lately where CPU cannot get ahead of GPU and all 20-50 us latencies arising even in cheap ops are exposed. I think sufficient condition is if no argument was expanded https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/fuser/executor.cpp#L110

Reply:
As far as I understood, this is hard to skip.